### DISCUSSION TITLE
Image getting darker with 'export_profile' => 'srgb'

### DISCUSSION THREAD

**Discussion by goldmerc** [libvips/php-vips#192] [Q&A]
I'm using vips 8.14.1 and thumbnail_buffer. Following this advice...

https://www.libvips.org/API/current/Using-vipsthumbnail.html

I'm passing in 'export_profile' => 'srgb'

I notice that the thumbnailed images look darker. If I remove 'export_profile' => 'srgb' the thumbnail colour matches the original perfectly.

What's happening here? I could just remove 'export_profile' => 'srgb' but I want to understand whats happening in case I'm going to get different results with different images.

Thanks (as always!)

**Comment by jcupitt** [comment]
Hi again @goldmerc,

Do you have a sample image which shows this problem?

**Comment by goldmerc** [comment]
Hi @jcupitt :-) 

Original Image...

![original-image](https://user-images.githubusercontent.com/1322182/221824715-91cb3372-6884-4fbc-b1ef-1558b13c5b50.jpg)

This is optimised using newFromFile. The colour stays the same...

![original-optimised](https://user-images.githubusercontent.com/1322182/221824995-06fe5217-09d2-4a3d-848d-c73217817e30.jpg)

But when it gets thumbnailed (thumbnail_buffer), it's darker... (same with jpg + webp)

![thumbnail](https://user-images.githubusercontent.com/1322182/221825077-21c5ab4f-14c9-4ebd-97ec-769fdf6d4018.jpg)

**Comment by goldmerc** [comment]
```
    'encoding_options' => [
        'jpg' => [
            'Q' => 85,
            'optimize_coding' => true,
            'interlace' => true,
            'strip' => true,
            'background' => "245 245 245",
        ],
        'webp' => [
            'Q' => 90,
            'strip' => true
        ],
    ]
```

**Comment by goldmerc** [comment]
If I remove 'export_profile' => 'srgb' from the thumbnail options, the colour stays as it should... (ie, not darker!)

![thumbnail-without-export_profile](https://user-images.githubusercontent.com/1322182/221826381-5a7b01c5-24f0-442a-b3a6-944c04f06c3b.jpg)

**Comment by jcupitt** [comment]
Huh that's weird, it looks like lcms isn't roundtripping correctly. If I do:

```
$ vips copy k2.jpg x.jpg[strip]
$ vips icc_transform x.jpg y.jpg srgb
```

They don't match, but they should. I don't have time to investigate now, unfortunately, maybe next week.

**Comment by goldmerc** [comment]
Sure, no rush. I'll leave it with you as it's all beyond me!

**Comment by jcupitt** [comment]
Looks like this could be related: https://github.com/libvips/libvips/issues/3150#issuecomment-1448514738

**Reply by kleisauke** [reply]
I confirm that using `cmsFLAGS_NOOPTIMIZE` would resolve this colour difference using the image in this discussion. Relevant PR: https://github.com/libvips/libvips/pull/3368

**Reply by goldmerc** [reply]
Great - thanks both. So, will this fix be included in the next release? And when might we see that?
Thanks again for your help.

**Reply by jcupitt** [reply]
I'd guess in libvips 8.15, maybe summer 2023.

**Reply by kleisauke** [reply]
PR https://github.com/libvips/libvips/pull/3373 targets the 8.14 branch, so this will be available in libvips 8.14.2 (RSN).
